Major Version update - see README for details.

peterjhill opened the pull request:
* Replaced the ssh library with [sshj](https://github.com/hierynomus/sshj)
    * Adds support for new ssh crypto algorithms
    * Adds support for FIPS compliant algorithms
    * More modern ssh implementation
* Added support for importing and building the library with maven
* Added FindBugs code testing to maven build

This is a major change to the netconf-java library. This version changes how Devices are
created. Instead of passing in parameters in a constructor, Device now uses a Builder. This
allows passing in whichver parameters you need and creating a unique Device object. Once created,
you no longer change change attributes like userName or port. You would need to create a new Device object.

See the Examples/CreateDevice class for examples on how to create a new Device.

vnitinv commented:
@peterjhill Thanks for the pull request. LICENSE looks to be made clear. It should be MIT or Apache (i assume). Let me discuss this with the team and do needful

peterjhill commented:
Excellent. I’m happy to leave the license terms to the lawyers. Cheers

ankitjain093 commented on src/main/java/net/juniper/netconf/Device.java:
```diff
@@ -0,0 +1,939 @@
+                throw new NetconfException(String.format("Error loading known hosts file: %s", e.getMessage()));
+            }
+            sshClient.setHostKeyRepository(sshClient.getHostKeyRepository());
+            log.info("Connecting to host {} on port {}.", hostName, port);
```
Could you please define the variable log

ankitjain093 commented on src/main/java/net/juniper/netconf/NetconfSession.java:
```diff
@@ -46,141 +46,192 @@
     }
     
     private void sendHello(String hello) throws IOException {
+        log.debug("Sending Netconf Capabilities Hello message. {}", hello);
```
Could you please define the log variable?

peterjhill commented:
Just added the maven javadoc plug in and configured it so that you can run
```mvn javadoc:javadoc```
and it will update the docs in the javadoc directory.

Also fixed all the javadoc strings that the javadoc validator found in error.

peterjhill commented:
During our integration testing, I saw that my .close() method only closed the channel and not the session,. Changed it to close both. Added private methods to check if either was open (though if the channel is closed, I sure hope the session is closed).

Pushed that up in a squashed commit to this PR

peterjhill commented:
added some tests for NetconfSession. not too many yet. added Mockito to mock out the back end ssh. Sort of want to add the Builder pattern for netconf session too. comment to vote.

peterjhill commented:
added more tests for the XML class.
